plugins/ingest/src/main/java/org/elasticsearch/plugin/ingest/transport/delete/DeletePipelineTransportAction.java

```diff
@@ -44,6 +44,8 @@ public DeletePipelineTransportAction(Settings settings, ThreadPool threadPool, T
 
     @Override
     protected void doExecute(DeletePipelineRequest request, ActionListener<DeletePipelineResponse> listener) {
+        // uboness: this logic belongs to the store. for example, have a
+        // `delete(String id, ActionListener<DeleteResponse>)` method on the store.
```

Comment:
agreed, but after doing this change, I realised that we need to do `delete(DeletePipelineRequest request, ActionListener<DeleteResponse>)` otherwise we lose the headers on the actual delete request. So I'll do this instead.